src/MarginTrading.Brokers/MarginTrading.BrokerBase/BrokerStartupBase.cs

```diff
@@ -0,0 +1,154 @@
+                .AddConsole(LogLevel.Error)
+                .AddDebug(LogLevel.Warning);
+
+            services.AddSingleton(loggerFactory);
+            services.AddLogging();
+            services.AddSingleton(Configuration);
+            services.AddMvc();
+
+            var settingsRoot = GetSettingsRoot();
+
+            var isLive = IsLive(Configuration);
+            Console.WriteLine($"IsLive: {isLive}");
+
+            var builder = new ContainerBuilder();
+            RegisterServices(services, settingsRoot, builder, isLive);
+            ApplicationContainer = builder.Build();
+
+            return new AutofacServiceProvider(ApplicationContainer);
+        }
+
+        protected virtual TSettingsRoot GetSettingsRoot()
+        {
+            TSettingsRoot settingsRoot;
+            string settingsLocation;
+            if (Environment.IsDevelopment())
+            {
+                settingsLocation = AppSettingsDevFile;
+                settingsRoot = Configuration.Get<TSettingsRoot>();
+            }
+            else
+            {
+                settingsLocation = Configuration["SettingsUrl"];
+                settingsRoot = SettingsProcessor.Process<TSettingsRoot>(settingsLocation.GetStringAsync().Result);
+            }
+
+            // ReSharper disable once ConditionIsAlwaysTrueOrFalse
+            if (settingsRoot == null)
+            {
+                throw new InvalidOperationException("Settings not found in " + settingsLocation);
+            }
+            return settingsRoot;
+        }
+
+        public virtual void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory,
+            IApplicationLifetime appLifetime)
+        {
+            app.UseMvc();
+
+            var application = app.ApplicationServices.GetService<IBrokerApplication>();
+
+            appLifetime.ApplicationStarted.Register(() => application.Run());
+            appLifetime.ApplicationStopping.Register(() => application.StopApplication());
+            appLifetime.ApplicationStopped.Register(() => ApplicationContainer.Dispose());
+        }
+
+        protected abstract void RegisterCustomServices(IServiceCollection services, ContainerBuilder builder,
+            TSettingsRoot settingsRoot, ILog log, bool isLive);
+
+        protected virtual ILog CreateLogWithSlack(IServiceCollection services, TSettingsRoot settings)
+        {
+            var logToConsole = new LogToConsole();
+            var logAggregate = new LogAggregate();
+
+            logAggregate.AddLogger(logToConsole);
+
+            var slackService = settings.SlackNotifications != null
+                ? services.UseSlackNotificationsSenderViaAzureQueue(settings.SlackNotifications.AzureQueue,
+                    logToConsole)
+                : null;
+
+            // Creating azure storage logger, which logs own messages to concole log
+            var dbLogConnectionString = settings.MtBrokersLogs?.DbConnString;
```

Comment:
All Demo and Live logs will be in one table and we could never understand which one is error source? May be at least use isLive property in table names?

Reply:
The connection strings to databases should be different then